Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -187,7 +222,8 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
         let channels = changes.compactMap { change -> ChatChannel? in
             switch change {
             case let .update(channel, _):
-                // We unlink the channels where `shouldListUpdatedChannel` returns `false`
+                // We unlink the channels that do not belong to this controller
+                guard channelBelongsToController(channel) else { return channel }
```

Comment:
Hm, my understanding is that if `filter` is present, the delegate should not be called 🤔 If I were a customer and move to the new `filter` API (to avoid deprecation warnings), I would provide the `filter` for custom query and remove overrides for `shouldAdd/shouldList`. But removing overrides from custom channel list VC would lead to default implementation in `ChatChannelVC`  being used where we check `channel.membership`. Or I'm missing smth?

Reply:
That's actually a good point